x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupResponseTranslator.java

```diff
@@ -187,48 +203,43 @@ public static SearchResponse translateResponse(MultiSearchResponse.Item[] rolled
      * @param msearchResponses The responses from the msearch, where the first response is the live-index response
      */
     public static SearchResponse combineResponses(MultiSearchResponse.Item[] msearchResponses,
-                                                  InternalAggregation.ReduceContext reduceContext) {
-        boolean liveMissing = false;
+                                                  InternalAggregation.ReduceContext reduceContext) throws Exception {
+
         assert msearchResponses.length >= 2;
 
-        // The live response is always first
-        MultiSearchResponse.Item liveResponse = msearchResponses[0];
-        if (liveResponse.isFailure()) {
-            Exception e = liveResponse.getFailure();
-            // If we have a rollup response we can tolerate a missing live response
-            if (e instanceof IndexNotFoundException) {
-                logger.warn("\"Live\" index not found during rollup search.", e);
-                liveMissing = true;
+        boolean first = true;
+        SearchResponse liveResponse = null;
+        List<SearchResponse> rolledResponses = new ArrayList<>();
+        for (MultiSearchResponse.Item item : msearchResponses) {
+            if (item.isFailure()) {
+                Exception e = item.getFailure();
+
+                // We can tolerate missing indices, assuming there is at least one to process at the end
+                if (e instanceof IndexNotFoundException) {
+                    String targetIndex = first ? "\"Live\"" : "Rollup";
+                    logger.warn(targetIndex + " index not found during rollup search.", e);
```

Comment:
Not sure we should log anything here, shouldn't this be the normal case when the live index is removed ? I also wonder if it is possible to get an `IndexNotFoundException` on the rollup search since we validate that the index exists before sending the query ?

Reply:
These `IndexNotFound` exceptions only arise if the index existed when the search request was executed (since we check local cluster state on the node to verify indices), but was deleted while the request was in-flight. 

It's essentially a partial-result because the user asked for `rollup, live` indices but we can only return `rollup` because of the deletion.  We don't really have a way to express partial results unless we co-opt the failed-shards metadata but that feels incorrect too.

I believe we ended up this way because, although it's a partial result, if the user executes the exact same request again we see there are no "live" indices and only query rollup.  The query will then work as Rollup will only execute against the rollup indices.

We could instead just be strict, fail the request and the user has to re-execute.

Reply:
> We could instead just be strict, fail the request and the user has to re-execute.

I think I prefer this option, just logging the partial result can be misleading and we have no way to convey the partial result information in the response so +1 to always throw an exception in this case and let the user re-execute;

Reply:
Works for me.  I'll try to word the exception so that the user knows it is a transient effect and they should try again.